File: android/app/src/main/kotlin/io/github/wikg1018/sitemark/CaptureTargetPolicy.kt
package io.github.wikg1018.sitemark

enum class RecoveryDisposition {
    CAPTURED,
    CANCELLED,
}

object CaptureTargetPolicy {
    private val safeCaptureId = Regex("^[A-Za-z0-9][A-Za-z0-9_-]{0,95}$")

    fun fileName(captureId: String): String {
        require(safeCaptureId.matches(captureId)) { "Invalid capture id" }
        return "$captureId.jpg"
    }

    fun recoveryDisposition(exists: Boolean, length: Long): RecoveryDisposition =
        if (exists && length > 0L) {
            RecoveryDisposition.CAPTURED
        } else {
            RecoveryDisposition.CANCELLED
        }
}

File: packages/sitemark_system_api/lib/src/ohos/capture_target_policy.dart
class CaptureTargetPolicy {
  static final RegExp _safeCaptureId = RegExp(
    r'^[A-Za-z0-9][A-Za-z0-9_-]{0,95}$',
  );

  static String fileName(String captureId) {
    if (!_safeCaptureId.hasMatch(captureId)) {
      throw ArgumentError.value(captureId, 'captureId', 'Invalid capture id');
    }
    return '$captureId.jpg';
  }

  static bool isCaptured({required bool exists, required int length}) {
    return exists && length > 0;
  }
}
